activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -141,6 +141,10 @@ def supports_datetime_with_precision?
         end
       end
 
+      def supports_virtual_columns?
+        !mariadb? && version >= "5.7.5"
```

Comment:
> One idea is not supporting virtual columns schema migrations at all

Agree this sets the clearest expectations which is probably the friendliest choice. 👍